src/circuit/circuit_builder.rs

```diff
@@ -892,19 +886,39 @@ where
         self.inner().parent.clone()
     }
 
-    /// Returns a reference to the cache associated with the circuit.
+    /// Lookup a value in the circuit cache or insert a value on cache miss.
     ///
     /// See [`cache`] module documentation for details.
-    pub(crate) fn cache(&self) -> RefMut<'_, CircuitCache> {
-        let circuit = self.inner_mut();
+    pub(crate) fn cache_lookup_or_insert_with<K, F>(&self, key: K, mut f: F) -> RefMut<'_, K::Value>
```

Comment:
```suggestion
    pub(crate) fn cache_get_or_insert_with<K, F>(&self, key: K, mut f: F) -> RefMut<'_, K::Value>
```
`.get_or_insert` is a more standard (and shorter) naming scheme

Reply:
but it's "cache lookup", not "cache getting".

Reply:
It's getting a value from the cache, right? That's what the return value is, at least